### PR TITLE
`gpstop -u` runs HUP commands locally when possible

### PR DESCRIPTION
**Pull request by nadeemg**
previously, `gpstop -u` used remote `ssh` for all hosts, all HUP commands. This PR changes to
add a conditional for when the host == localhost, and runs the command
without `ssh`. 

MOTIVATION
https://github.com/moby/moby/pull/36466 fixes an issue with AppArmor in Docker. It is not available yet in Kubernetes. It causes *different* permissions when attempting to send a signal like HUP from `ssh` vs. from localhost. This PR helps work around the issue.

### CODE REVIEW

File: gpMgmt/bin/gpstop [hunk 1]
```diff
@@ -743,9 +744,15 @@ class GpStop:
         self.pool = SighupWorkerPool(numWorkers=workers)
         dbList = self.gparray.getDbList()
         dispatch_count = 0
+        hostname = socket.gethostname()
         logger.info("Signalling all postmaster processes to reload")
         for db in dbList:
-            cmd = pg.ReloadDbConf(name="reload segment number " + str(db.getSegmentDbId())
+            if db.getSegmentHostName() == "localhost" or db.getSegmentHostName() == hostname:
```
Comment (jchampio) [inline gpMgmt/bin/gpstop]: Do we need to worry about any other aliases (or IP addresses) for the localhost?

File: gpMgmt/bin/gpstop [hunk 2]
```diff
@@ -743,9 +744,15 @@ class GpStop:
...
+                cmd = pg.ReloadDbConf(name="reload segment number " + str(db.getSegmentDbId())
+                                      , db=db
+                                      )
+            else:
+                cmd = pg.ReloadDbConf(name="reload segment number " + str(db.getSegmentDbId())
```
Comment (jchampio) [inline gpMgmt/bin/gpstop]: We should be able to just do a single call to `ReloadDbConf()` here, and set `ctxt` and `remoteHost` depending on whether the host is local or not.

File: gpMgmt/bin/gpstop (hunk 1 above)
Comment (larham) [reply gpMgmt/bin/gpstop]: This question revolves around how the Greenplum cluster was defined. `db.getSegmentHostName()` returns the value stored in table `gp_segment_configuration`.  A user could create DNS aliases for the Master (or any other node) and use that alias to configure Greenplum, such that hostname != alias. Is there a way for us to know all possible aliases?  

For the key use-case on Kubernetes, there is no unexpected alias, and the hostname does indeed match the names used to define the cluster, so this code works in that case. 

If there is no match, the code still does the legacy "remote" operation, no change from previous behavior.

File: gpMgmt/bin/gpstop (hunk 2 above)
Comment (larham) [reply gpMgmt/bin/gpstop]: Done.

File: gpMgmt/bin/gpstop (hunk 1 above)
Comment (jchampio) [reply gpMgmt/bin/gpstop]: `gethostbyaddr` will give you some "aliases", but I'm not sure it's actually the list of aliases you want for this use case... I think the only reliable source of truth here is probably the DNS system itself.

This approach works for the desired use case, so I'm not too worried about it. If we run into trouble with it later, maybe we can instead check `self.dburl.pghost`, which should have the hostname of our current connection to the master. That's probably more likely to be set to `localhost` even if the master hostname in the catalog isn't, or at least to the value of `PGHOST` (which is _probably_ not going to be a qualified domain name if the dbadmin expects to be connecting to the local machine). And then, if any segment host name matches the master host name in the catalog, we know we can try direct execution there too. But that is *way* too overengineered to worry about now. :smile:

**Pull request merged**